.github/workflows/ci.yml

```diff
@@ -2,58 +2,112 @@ name: CI
 on:
   - push
   - pull_request
+
 jobs:
-  build_and_test:
-    name: many-framework build and tests
+  build:
+    name: ${{ matrix.make.name }} (${{ matrix.os }})
     runs-on: ${{ matrix.os }}
     strategy:
+      fail-fast: false
       matrix:
-        os: [ubuntu-22.04]
+        os: [ubuntu-latest, macos-latest]
+        rust: [nightly]
+        make:
+          - name: Clippy
```

Comment:
This task also checks `fmt`. Either reflect it in the name/makefile or create another entry.